src/vs/workbench/services/telemetry/browser/telemetryService.ts

```diff
@@ -65,7 +61,7 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 		remoteAgentService: IRemoteAgentService
 	) {
 		const telemetrySupported = supportsTelemetry(productService, environmentService) && productService.aiConfig?.ariaKey;
-		if (telemetrySupported && this.impl === NullTelemetryService && this.telemetryLevel.value !== TelemetryLevel.NONE) {
+		if (telemetrySupported && getTelemetryLevel(configurationService) !== TelemetryLevel.NONE && this.impl === NullTelemetryService) {
```

Comment:
@jeanp413 Trying to understand what changed here. `this.telemetryLevel.value` is actually a more robust version of `getTelemetryLevel` as it also respects things like the `prouct.json`

Here's the base one 
https://github.com/microsoft/vscode/blob/bc58fb0e14c33a781ee7cb48c3f14fd33f13110e/src/vs/platform/telemetry/common/telemetryService.ts#L71-L83

Here's the null one which just returns `NONE`
https://github.com/microsoft/vscode/blob/bc58fb0e14c33a781ee7cb48c3f14fd33f13110e/src/vs/platform/telemetry/common/telemetryUtils.ts#L36